test/spectral/tests/custom-rules/prohibit-sentence-style-summaries.test.js

```diff
@@ -0,0 +1,102 @@
+const inCodeValidator = require('../../../../src/lib');
+
+describe('spectral - summaries do not have trailing period', () => {
+  it('should not display error when summary does not have trailing period', async () => {
+    const spec = {
+      openapi: '3.0.0',
+      info: {
+        version: '1.0.0',
+        title: 'Summary does not have trailing comma'
+      },
+      servers: {},
+      paths: {
+        createTrailingPeriod: {
+          post: {
+            operationId: 'createTrailingPeriod',
+            description: 'creates trailing period',
+            summary: 'No trailing period here'
+          }
+        }
+      }
+    };
+
+    const result = await inCodeValidator(spec, true);
+    const relevantError = result.warnings.find(
+      v => v.rule === 'prohibit-summary-sentence-style'
+    );
+    expect(relevantError).toBeUndefined();
+  });
+
+  it('should display error when a summary has trailing period', async () => {
+    const spec = {
+      openapi: '3.0.0',
+      info: {
+        version: '1.0.0',
+        title: 'Summary has trailing period'
+      },
+      servers: {},
+      paths: {
+        createTrailingPeriod: {
+          post: {
+            operationId: 'createTrailingPeriod',
+            description: 'creates trailing period',
+            summary: 'There is a trailing period here.'
+          }
+        }
+      }
+    };
+
+    const validationResult = await inCodeValidator(spec, true);
+    expect(validationResult.errors).not.toBeUndefined();
+    const error = validationResult.warnings.find(
+      v => v.rule === 'prohibit-summary-sentence-style'
+    );
+    expect(error).not.toBeUndefined();
+    const errorPath = ['paths', 'createTrailingPeriod', 'post', 'summary']
+      .sort()
+      .join('');
```

Comment:
Is there a reason you sorted and joined here rather than checking for the explicit order the path was returned in like you do for the last test? I think this is fine but maybe not as robust as strictly checking the actual order

Reply:
This was an initial solution for checking equality of two arrays. But @barrett-schonefeld showed me what help `jest` can offer here. I refactored this test too.